src/sentry/static/sentry/app/components/events/eventTags.tsx

```diff
@@ -42,7 +42,10 @@ class EventTags extends React.Component<EventTagsProps> {
         query.project = tag.value;
         break;
       default:
-        query.query = `${query.query} ${tag.key}:${tag.value}`;
+        query.query =
+          tag.value.indexOf(' ') > -1
+            ? `${query.query} ${tag.key}:"${tag.value}"`
```

Comment:
I don't remember how (if any) we handle escaping quotes - can you dig a bit into that as a follow-up?

Reply:
They need to have `\` in front of the `"`. https://github.com/getsentry/sentry/blob/80b7a59635efcba64d1c568ee98429bbdd7e6327/src/sentry/api/event_search.py#L75-L86

You could always wrap & escape tag values to be on the safe side.